src/backend/parser/gram.y

```diff
@@ -16730,8 +16764,38 @@ c_expr:		columnref								{ $$ = $1; }
 
 scatter_clause:
 		/* EMPTY */						{ $$ = NIL; }
-		| SCATTER RANDOMLY				{ $$ = list_make1(NULL); }
-		| SCATTER BY expr_list 			{ $$ = $3; }
+		| SCATTER RANDOMLY
+		  {
+			  /*
+			   * In singlenode mode, scatter clause has no real effect.
+			   * We simply ignore it and issue a warning to ensure compatibility.
+			   */
+			  if (GP_ROLE_SINGLENODE == Gp_role)
+			  {
+			  	ereport(WARNING,
+			  			(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+			  				errmsg("SCATTER BY clause has no effect in singlenode mode")));
+			  	$$ = NIL;
+			  }
+			  else
+			  {
+			  	$$ = list_make1(NULL);
+			  }
+		  }
+		| SCATTER BY expr_list
+		  {
+			  if (GP_ROLE_SINGLENODE == Gp_role)
+			  {
+			  	ereport(WARNING,
```

Comment:
consider put warning in `transformDistributedBy` , but not in gram.y ?

Reply:
What's the benefit? I think it's an issue in grammar level. Like we don't support this grammar.

Reply:
like `DISTRIBUTED BY` warning 
```
postgres=# CREATE table x (a int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
```

Reply:
Actually, I don't get why gpdb don't write it into gram.y.😂 From my point of view, this part of logic should be done in grammar level.

```
OptTemp:	TEMPORARY					{ $$ = RELPERSISTENCE_TEMP; }
			| TEMP						{ $$ = RELPERSISTENCE_TEMP; }
			| LOCAL TEMPORARY			{ $$ = RELPERSISTENCE_TEMP; }
			| LOCAL TEMP				{ $$ = RELPERSISTENCE_TEMP; }
			| GLOBAL TEMPORARY
				{
					ereport(WARNING,
							(errmsg("GLOBAL is deprecated in temporary table creation"),
							 parser_errposition(@1)));
					$$ = RELPERSISTENCE_TEMP;
				}
			| GLOBAL TEMP
				{
					ereport(WARNING,
							(errmsg("GLOBAL is deprecated in temporary table creation"),
							 parser_errposition(@1)));
					$$ = RELPERSISTENCE_TEMP;
				}
			| UNLOGGED					{ $$ = RELPERSISTENCE_UNLOGGED; }
			| /*EMPTY*/					{ $$ = RELPERSISTENCE_PERMANENT; }
		;
```
And I found that in gram.y there are similar warnings.🤔 So, what is the case you think it should be written in gram.y and what's the "not clean" part of the change in gram.y. 
I'm still on my side now.

Reply:
That's a fair point. I have a better way to change it in gram.y. Let me change it first.

Reply:
Updated.